pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -182,6 +155,106 @@ final class Event {
           'name': name,
         };
 
+  /// Event to capture previous custom dimension values from
+  /// legacy flutter analytics.
+  ///
+  /// The parameters below are a superset of all the implementations
+  /// of the `usageValues` getter in the `FlutterCommand` class within
+  /// the flutter-tool. There should never be a time where all of the parameters
+  /// are passed to this constructor.
+  ///
+  /// The custom dimensions from each implementation are grouped by sharing
+  /// the same prefix. The [workflow] indicates which implementation is being
+  /// sent, for example, "create".
+  Event.commandUsageValues({
+    required String workflow,
+    required bool commandHasTerminal,
+
+    // Assemble && build bundle implementation parameters
+    String? buildBundleTargetPlatform,
+    bool? buildBundleIsModule,
+
+    // Build aar implementation parameters
+    String? buildAarProjectType,
+    String? buildAarTargetPlatform,
+
+    // Build apk implementation parameters
+    String? buildApkTargetPlatform,
+    String? buildApkBuildMode,
+    bool? buildApkSplitPerAbi,
+
+    // Build app bundle implementation parameters
+    String? buildAppBundleTargetPlatform,
+    String? buildAppBundleBuildMode,
+
+    // Create implementation parameters
+    String? createProjectType,
+    String? createAndroidLanguage,
+    String? createIosLanguage,
+
+    // Packages implementation parameters
```

Comment:
I vaguely remember a conversation where we looped in Sigurd on whether or not we need to keep these in the Flutter tool. What was the conclusion there?

Reply:
We ended up deciding to deprecate that event since pub will also have their own implementation of this package. But we didn't get any response from Sigurd to confirm it or not

- https://github.com/dart-lang/tools/pull/194#discussion_r1385453724

I think it makes sense to also deprecate here if we did it there, wdyt?

Reply:
chatted offline, we can decide whether or not we deprecate these independent of this PR

Reply:
SGTM